### PR TITLE
Replace References to Log Entity with New BuildStep Entity

### PR DESCRIPTION
**Pull request by samuelsogbesan**
This PR integrates a new Entity that represents the step-by-step information for a given build. 
This change comes as a result of a decision to **no longer source the build logs from the top-level "logs" field**, but instead from **the "log" field attached to each child of the steps field**.

#### Justifications for change
- The steps field provides several other pieces of supporting meta-data on top of the logs.
- The logs field doesn't give enough detail to the purpose or meaning of the logs.
- There is not a one-to-one relationship with steps and logs. The steps field helps us identify which logs belong to which step.

Points of contention:
- What should the name of the Entity via the @Entity annotation?

### CODE REVIEW

**Review by holounic — APPROVED** [review]
LGTM

File: src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public ParsedBuildbotData(String commitHash, String builderName,
    * @return true if all required fields are available, false if not.
    */
   boolean validUpdateData() {
-    return (commitHash != null && builderName != null && logs != null && status != null);
```
Comment (MagoDelBlocco) [inline src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:62]: Please keep in mind when merging that both validUpdateData and toBuilder have been deprecated

Comment (samuelsogbesan) [reply src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:62]: Thanks. Is the deprecation formally communicated in another PR? Should I manually remove the methods? @MagoDelBlocco

Comment (MagoDelBlocco) [reply src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:62]: It has been done in #79. I left this as a heads-up for the merging process.

Comment (samuelsogbesan) [reply src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:62]: Cool, thanks for the heads-up.

File: src/main/java/com/google/graphgeckos/dashboard/storage/BuildStep.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  private final boolean isFinished;
+
+  // Indicator of whether or not the step has started
+  private final boolean isStarted;
+
+  // All the logs associated with this step
+  private final <List>Log logs;
+
+  /**
+   * Constructs an instance of BuildStep. name, text and logs can be null.
```
Comment (MagoDelBlocco) [inline src/main/java/com/google/graphgeckos/dashboard/storage/BuildStep.java:45]: `Constructs an instance of BuildStep` is redundant.

File: src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java [hunk 3]
```diff
@@ -26,34 +26,35 @@
  * Fields included:
```
Comment (MagoDelBlocco) [inline src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:26]: ParsedBuildbotData got superseded in #88 . Consider rebasing on that.

File: src/main/java/com/google/graphgeckos/dashboard/storage/BuildStep.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    this.text = Arrays.toString(text);
+    this.isFinished = isFinished;
+    this.isStarted = isStarted;
+    this.logs = Arrays.asList(logs);
+  }
+
+  /**
+   * Returns the name of the log as a String. Can be null.
+   */
+  public String getName() {
```
Comment (MagoDelBlocco) [inline src/main/java/com/google/graphgeckos/dashboard/storage/BuildStep.java:58]: Consider adding `@NonNull` [annotation](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/lang/NonNull.html#:~:text=Annotation%20Type%20NonNull&text=A%20common%20Spring%20annotation%20to,infer%20nullability%20of%20Spring%20API.) for the return types.

**Review by MagoDelBlocco — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/graphgeckos/dashboard/storage/BuildStep.java (hunk 4 above)
Comment (samuelsogbesan) [reply src/main/java/com/google/graphgeckos/dashboard/storage/BuildStep.java:63]: I was under the assumption name could be null here!

Comment (samuelsogbesan) [reply src/main/java/com/google/graphgeckos/dashboard/storage/BuildStep.java:63]: I don't have a counter example so i'll add the annotation.

File: src/main/java/com/google/graphgeckos/dashboard/storage/BuildStep.java (hunk 2 above)
Comment (samuelsogbesan) [reply src/main/java/com/google/graphgeckos/dashboard/storage/BuildStep.java]: Updated in a previous commit.

File: src/main/java/com/google/graphgeckos/dashboard/storage/BuildStep.java (hunk 4 above)
Comment (samuelsogbesan) [reply src/main/java/com/google/graphgeckos/dashboard/storage/BuildStep.java:64]: Annoted with commit 6d47eb4

File: src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java (hunk 3 above)
Comment (samuelsogbesan) [reply src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:26]: Given that the file was removed, will @holounic need to integrate these changes onto the <code>Builder</code> class?

Comment (MagoDelBlocco) [reply src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java:26]: Yes

Comment (samuelsogbesan) [reply src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java]: Pulled origin here. Updated and moved files accordingly.

Comment (samuelsogbesan) [reply src/main/java/com/google/graphgeckos/dashboard/storage/ParsedBuildbotData.java]: Pulled origin here. Updated and moved files accordingly.

**Review by MagoDelBlocco — APPROVED** [review]

**Comment by MagoDelBlocco** [comment]
> What should the name of the Entity via the @entity annotation?

I'd say "buildstep"

**Pull request closed without merging**